src-electron/generator/helper-endpointconfig.js

```diff
@@ -651,15 +651,14 @@ function determineAttributeDefaultValue(
   typeSize,
   isNullable
 ) {
-  if (specifiedDefault !== null || !isNullable) {
+  if (specifiedDefault && (specifiedDefault !== null || !isNullable)) {
```

Comment:
I don't understand this code.  If `specifiedDefault` is truthy, then certainly `specifiedDefault !== null` is true, and then it doesn't matter what's going on with `isNullable`, so why do we have `isNullable` at all?

Reply:
Sometimes "specifiedDefault" is not truthy AND not null. Sometimes it's in an empty state.

Reply:
For example, when it's in this "empty" state it will fail tests checking if it's "undefined" or "null" because it's something else

Reply:
> Sometimes "specifiedDefault" is not truthy AND not null. Sometimes it's in an empty state.

Yes, sure.  Lots of falsy things in JS are not null: `false`, `0`, `""`, `undefined`.

My poiunt is, this condition as written is completely identical to ` if (specifiedDefault) {`, so why is it not written that way?

And if the input is in fact 0, do we really want to output `0xFF` and whatnot?

Reply:
@bzbarsky-apple thanks for your review. I reverted the commit and will make a new PR.

Reply:
@bzbarsky-apple when it comes to this code, I was under the impression that the javascript representation of specifiedDefault being 0 would be a string which would return true 

That being said, I reverted this because if there is an empty string then it will return false which as discussed on slack is a bug